### PR TITLE
Add documentation for integrating R2

### LINKED ISSUE #310: Support R2 in Delta Sharing OSS Reference Server

### PR DESCRIPTION
**Pull request by zhu-tom**
Add documentation for integrating Cloudflare R2. This implementation uses the R2 implementation of the S3 API and instructs on how to specify proper configurations to connect to R2. 

Closes #310

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -457,6 +457,34 @@ export GOOGLE_APPLICATION_CREDENTIALS="KEY_PATH"
 
 Replace `KEY_PATH` with path of the JSON file that contains your service account key.
 
+### Cloudflare R2
```
Comment (wchau) [inline README.md:460]: Should mention that it should also use `s3a://`

Comment (zhu-tom) [reply README.md:460]: I have the example in `delta-sharing-server.yaml.template`, should I also include it here? Even if the section seems to be mainly focused on credentials

Comment (wchau) [reply README.md:460]: Yes. A brief note should be enough. It was also mentioned above in the `s3` section.

Comment (zhu-tom) [reply README.md:460]: Added 👍

**Review by wchau — APPROVED** [review]

**Pull request merged**